dace/frontend/python/newast.py

```diff
@@ -2095,8 +2099,12 @@ def _recursive_visit(self, body: List[ast.AST], name: str, lineno: int, last_sta
             self.globals.update(extra_symbols)
 
         # Recursive loop processing
+        return_stmt = False
         for stmt in body:
             self.visit_TopLevel(stmt)
+            if isinstance(stmt, ast.Return):
```

Comment:
same as above

Reply:
Here we need to know if there was a return statement, in order to create the appropriate edges in if-scopes. The "break" statement itself should be unecessary though. Removed.

Reply:
Added description in the docstring.